server/src/main/java/org/opensearch/extensions/ExtensionActionListenerOnFailureRequest.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.extensions;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+
+import java.io.IOException;
+import java.util.Objects;
+
+/**
+ * CLusterService Request for Action Listener onFailure
+ *
+ * @opensearch.internal
+ */
+public class ExtensionActionListenerOnFailureRequest extends ExtensionRequest {
+    private static final Logger logger = LogManager.getLogger(ExtensionRequest.class);
+    private ExtensionsOrchestrator.RequestType requestType;
+    private String failureException;
+
+    public ExtensionActionListenerOnFailureRequest(ExtensionsOrchestrator.RequestType requestType, String failureException) {
+        super(requestType);
```

Comment:
We don't need `requestType` here. It was added for checking the type in the `ExtensionRequest `

Reply:
This also means to remove the `extends`.